src/array/kernel.cc

```diff
@@ -296,61 +271,81 @@ DGL_REGISTER_GLOBAL("sparse._CAPI_DGLKernelGetEdgeMapping")
+    const HeteroGraphPtr B = B_ref.sptr();
+    CHECK_EQ(A->NumEdgeTypes(), 1) << "Graph must have one edge type.";
+    CHECK_EQ(A->NumEdgeTypes(), 1) << "Graph must have one edge type.";
+    CHECK_EQ(A->NumVertices(A->NumVertexTypes() - 1), B->NumVertices(0)) <<
+      "Number of vertices do not match.";
+    const auto A_csr = A_ref.sptr()->GetCSRMatrix(0);
+    const auto B_csr = B_ref.sptr()->GetCSRMatrix(0);
+    CHECK_EQ(A_csr.num_cols, B_csr.num_rows) << "Number of nodes mismatch.";
+    auto result = CSRMM(A_csr, A_weights, B_csr, B_weights);
+
+    List<ObjectRef> ret;
+    ret.push_back(HeteroGraphRef(CreateFromCSR(num_vtypes, result.first, ALL_CODE)));
     ret.push_back(Value(MakeValue(result.second)));
     *rv = ret;
   });
 
 DGL_REGISTER_GLOBAL("sparse._CAPI_DGLCSRSum")
 .set_body([] (DGLArgs args, DGLRetValue* rv) {
-    int M = args[0];
-    int N = args[1];
-    List<Value> A_indptr = args[2];
-    List<Value> A_indices = args[3];
-    List<Value> A_data = args[4];
-    std::vector<NDArray> weights = ListValueToVector<NDArray>(A_data);
-    std::vector<CSRMatrix> mats(A_indptr.size());
-    for (int i = 0; i < A_indptr.size(); ++i)
-      mats[i] = CSRMatrix(M, N, A_indptr[i]->data, A_indices[i]->data);
+    List<HeteroGraphRef> A_refs = args[0];
+    List<Value> A_weights = args[1];
+
+    std::vector<NDArray> weights = ListValueToVector<NDArray>(A_weights);
+    std::vector<CSRMatrix> mats;
+    mats.reserve(A_refs.size());
+    int num_vtypes = 0;
+    for (auto A_ref : A_refs) {
+      const HeteroGraphPtr A = A_ref.sptr();
+      CHECK_EQ(A->NumEdgeTypes(), 1) << "Graphs must have only one edge type.";
+      mats.push_back(A->GetCSRMatrix(0));
+      if (num_vtypes == 0)
+        num_vtypes = A->NumVertexTypes();
+    }
     auto result = CSRSum(mats, weights);
-    List<Value> ret;
-    ret.push_back(Value(MakeValue(result.first.indptr)));
-    ret.push_back(Value(MakeValue(result.first.indices)));
+
+    List<ObjectRef> ret;
+    ret.push_back(HeteroGraphRef(CreateFromCSR(num_vtypes, result.first, ALL_CODE)));
     ret.push_back(Value(MakeValue(result.second)));
     *rv = ret;
   });
 
 DGL_REGISTER_GLOBAL("sparse._CAPI_DGLCSRMask")
 .set_body([] (DGLArgs args, DGLRetValue* rv) {
-    int M = args[0];
-    int N = args[1];
-    NDArray A_indptr = args[2];
-    NDArray A_indices = args[3];
-    NDArray A_data = args[4];
-    NDArray B_indptr = args[5];
-    NDArray B_indices = args[6];
-    auto result = CSRMask(
-        CSRMatrix(M, N, A_indptr, A_indices),
-        A_data,
-        CSRMatrix(M, N, B_indptr, B_indices));
+    const HeteroGraphRef A_ref = args[0];
+    NDArray A_weights = args[1];
+    const HeteroGraphRef B_ref = args[2];
+
+    const HeteroGraphPtr A = A_ref.sptr();
+    const HeteroGraphPtr B = B_ref.sptr();
+    CHECK_EQ(A->NumEdgeTypes(), 1) << "Graph must have only one edge type.";
+    CHECK_EQ(B->NumEdgeTypes(), 1) << "Graph must have only one edge type.";
```

Comment:
Also check A and B have the same (1) number of nodes, (2) idtype and (3) device context.

Reply:
IDType and context are already checked in `src/array/array.cc`.